core/src/main/scala/org/apache/spark/executor/ExecutorMetrics.scala

```diff
@@ -28,16 +28,14 @@ import org.apache.spark.metrics.ExecutorMetricType
 @DeveloperApi
 class ExecutorMetrics private[spark] extends Serializable {
 
-  // Metrics are indexed by ExecutorMetricType.values
-  private val metrics = new Array[Long](ExecutorMetricType.values.length)
-
+  private val metrics = new Array[Long](ExecutorMetricType.numMetrics)
```

Comment:
I'd keep a comment here explaining this array, just update that its now indexed by `ExecutorMetricType.metricToOffset`